evalml/pipelines/components/component_base.py

```diff
@@ -80,6 +94,10 @@ def default_parameters(cls):
 
         return cls._default_parameters
 
+    @classproperty
+    def _supported_by_list_API(cls):
```

Comment:
If the plan is move ahead with deprecating the list api for pipelines, I'm wondering if we even need this anymore?

It's currently only used in `make_pipeline` and it only called on the components from `_get_preprocessing_components`. That's an implementation we control - so those components should always support the list api right?

I guess what I'm trying to get at is that we decide to get rid of the list api, then the only "list to dict" conversion that would exist is in `make_pipeline` but that's only because `_make_preprocessing_components` was written way before the `ComponentGraph` abstraction! If you're with me so far, then I think what we should do is have `_make_preprocessing_components` generate a component graph dict directly. 

That being said, I think this suggestion would only apply if we decide to deprecate the list api completely. If we go with the "half-half" solution from last week, I think the way you have this laid out currently is good!

I'm curious what you think about all this. It seems we're still on the fence about whether to get rid of the list api? 😅 Maybe this is something to broach with more members of the team?